cs/universegenerator.go

```diff
@@ -297,83 +297,90 @@ func (ug *universeGenerator) generatePlayerHomeworlds(area Vector) error {
-					distToHomeworld := planet.Position.DistanceSquaredTo(homeworld.Position)
-					if distToHomeworld <= closestDistance {
-						closestDistance = distToHomeworld
+					
+					// if we can't find a planet within tolerances, pick the farthest one
+					shortedDistanceToPlanets := planet.shortestDistanceToPlanets(&ownedPlanets)
+					if shortedDistanceToPlanets >= farthestDistance {
+						farthestDistance = shortedDistanceToPlanets
 						playerPlanet = planet
 					}
-					if distToHomeworld <= float64(rules.MaxExtraWorldDistance*rules.MaxExtraWorldDistance) && distToHomeworld >= float64(rules.MinExtraWorldDistance*rules.MinExtraWorldDistance) {
+					if len(ownedPlanets) == 0 || shortedDistanceToPlanets > minPlayerDistance {
 						playerPlanet = planet
 						break
 					}
 				}
-
+				homeworld = playerPlanet
+			
 			} else {
-				// homeworld should be distant from other players
+				// extra planets are close to the homeworld
 				for _, planet := range ug.universe.Planets {
 					if planet.Owned() {
 						continue
 					}
 
-					// if we can't find a planet within tolerances, pick the farthest one
-					shortedDistanceToPlanets := planet.shortestDistanceToPlanets(&ownedPlanets)
-					if shortedDistanceToPlanets >= farthestDistance {
-						farthestDistance = shortedDistanceToPlanets
+					// if we can't find a planet within tolerances, pick the closest one
+					distToHomeworld := planet.Position.DistanceSquaredTo(homeworld.Position)
+					if distToHomeworld <= closestDistance {
+						closestDistance = distToHomeworld
 						playerPlanet = planet
 					}
-					if len(ownedPlanets) == 0 || shortedDistanceToPlanets > minPlayerDistance {
+					if distToHomeworld <= float64(rules.MaxExtraWorldDistance*rules.MaxExtraWorldDistance) && distToHomeworld >= float64(rules.MinExtraWorldDistance*rules.MinExtraWorldDistance) {
 						playerPlanet = planet
 						break
 					}
 				}
-
-				homeworld = playerPlanet
 			}
-
+			
 			if playerPlanet == nil {
 				return fmt.Errorf("find homeworld for player %v among %d planets, minDistance: %0.1f", player, len(ug.universe.Planets), minPlayerDistance)
 			}
 
 			ownedPlanets = append(ownedPlanets, playerPlanet)
 
-			// our starting planet starts with default fleets
-			surfaceMinerals := homeworldSurfaceMinerals
-			if startingPlanetIndex != 0 {
-				surfaceMinerals = extraWorldSurfaceMinerals
+			var surface Mineral
+			if startingPlanet.Homeworld {
+				surface = homeworldSurfaceMinerals
+			} else {
+				surface = extraWorldSurfaceMinerals
 			}
 
 			// make a new starter world
-			playerPlanet.initStartingWorld(player, &ug.Rules, startingPlanet, homeworldMinConc, surfaceMinerals)
+			playerPlanet.initStartingWorld(player, &ug.Rules, startingPlanet, homeworldMinConc, surface)
+			if !startingPlanet.Homeworld && !ug.MaxMinerals {
+				// starting planets start with different mincons
+				playerPlanet.MineralConcentration = randomizeMinerals(rules, playerPlanet.Hab.Rad)
```

Comment:
we can't randomize this, all extra worlds should be the same. If two players player IT, one shouldn't get an extra world with bum minerals.

Reply:
Well, OG stars! randomized extra world mincons & hab.

Reply:
ah, so they do. nm!